shared/cluster.go

```diff
@@ -347,21 +363,32 @@ func parsePods(res string) []Pod {
 			continue
 		}
 
-		if len(fields) < 9 {
+		if len(fields) == 9 {
```

Comment:
can we improve this bunch of elseif , else and if ? 

idk maybe creating an offset with the minimun fields allowed and adding if it is a namespace or not

Reply:
also sorry, why not counting with namespace sometimes?

Reply:
> can we improve this bunch of elseif , else and if ?
Probably! I didn't want to break what was already there. I'm not sure why there was the check in the first place and didn't really want to dig into it too hard as part of these changes.

> why not counting with namespace sometimes?
In the case of GetPodsByNamespace, doing like a `kubectl get pods -n somenamespace ...` the output doesn't have the NAMESPACE column at all, so this accounts for that. And in the case that function is being used, the namespace should already be known anyway so it's okay if it's not in the output.

Reply:
got it tk for the explation

but tbh in this case we should totally improve that, sorry! you added a lot of more conditionals and code duplication

Reply:
> Probably! I didn't want to break what was already there. I'm not sure why there was the check in the first place and didn't really want to dig into it too hard as part of these changes.

that check was to validate the range index, to never break if its less then 9 ( total of fields that we were expecting to deal with  +.1 )  
its a way to ensure that parsing above would not break ( which already happen due to the nature of parsing outputs from cli )

Reply:
in those cases a good approach would working with offsets 
that way u dont need else even more elsif and avoid duplication.

tbh i dont have something done on my mind but something like: 
```
offset == 0

if len == {exact number when expecting NS} {
NamesPace = [0]   because its the first field right 
offset= 1
}
        Name    offset +1 
 
and then follow through



```

Reply:
Addressed with https://github.com/rancher/distros-test-framework/pull/40/commits/780acf3dc2bd5387e112dffc3ba3dd2ae96cec5e but I kind of like the if/else better myself. Let me know what you (and others) think here and we can use this approach if desired!

Reply:
let others give their opinion, for me makes no sense adding else, if, else if unnecessary. 
also duplicating code.